glean-core/src/system.rs

```diff
@@ -0,0 +1,81 @@
+// Copyright (c) 2017 The Rust Project Developers
+// Licensed under the MIT License.
+// Original license:
+// https://github.com/RustSec/platforms-crate/blob/ebbd3403243067ba3096f31684557285e352b639/LICENSE-MIT
+//
+// Permission is hereby granted, free of charge, to any
+// person obtaining a copy of this software and associated
+// documentation files (the "Software"), to deal in the
+// Software without restriction, including without
+// limitation the rights to use, copy, modify, merge,
+// publish, distribute, sublicense, and/or sell copies of
+// the Software, and to permit persons to whom the Software
+// is furnished to do so, subject to the following
+// conditions:
+//
+// The above copyright notice and this permission notice
+// shall be included in all copies or substantial portions
+// of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF
+// ANY KIND, EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED
+// TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A
+// PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT
+// SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY
+// CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION
+// OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR
+// IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER
+// DEALINGS IN THE SOFTWARE.
+
+//! Detect and expose `target_os` as a constant.
+//!
+//! Code adopted from the "platforms" crate: <https://github.com/RustSec/platforms-crate>.
+
+#[cfg(target_os = "android")]
+/// `target_os` when building this crate: `android`
+pub const OS: &str = "Android";
+
+#[cfg(target_os = "ios")]
+/// `target_os` when building this crate: `ios`
+pub const OS: &str = "iOS";
+
+#[cfg(target_os = "linux")]
+/// `target_os` when building this crate: `linux`
+pub const OS: &str = "Linux";
+
+#[cfg(target_os = "macos")]
+/// `target_os` when building this crate: `macos`
+pub const OS: &str = "Darwin";
```

Comment:
cc @chutten: I guess `Darwin` is the right thing. that's what we currently have in Telemetry as well.

Reply:
@badboy I think we should talk to @fbertsch about this, too. The `telemetry.main` dataset, for example, has the `normalized_os` field that reports `"Mac"`. Maybe we can do the proper thing here by default?

Reply:
Yeah, if we were interested in keeping what Telemetry had, then windows should be Windows_NT : |

Let's not slow down too much with consultation, though, because we're not going to be sending any data with this info for a while. If we don't get an answer, say, today then we can check in a "we'd be okay with this if it shipped" solution and file a follow-up to consult on the exact value DE and DS might want.

Reply:
I'll informally ask Frank for a quick answer.

Reply:
I'm fine with not blocking on this, now, but please do file a follow-up bug for understanding what's the best value we can offer here.

Reply:
https://bugzilla.mozilla.org/show_bug.cgi?id=1626301